File: python/spot.py
import asyncio
import discord

class Spot(discord.Client):
    '''
    import spot
    with spot(id, token) as spot:
        spot.message('i love you')
    '''
    def __init__(self, user_id, token, loop=asyncio.new_event_loop(), *args, **kwargs):
        self.loop = loop
        super().__init__(*args, **kwargs)
        async def login(token):
            await self.login(token)
            user = await self.fetch_user(user_id)
            self.channel = await user.create_dm()
            await self.channel.send('On duty!')
        self.loop.run_until_complete(login(token))

    def __enter__(self):
        return self

    def exit(self):
        async def goodbye():
            await self.channel.send('Work finished.')
            await self.logout()
        self.loop.run_until_complete(goodbye())

    def __exit__(self, *args):
        self.exit()
    
    def message(self, text):
        self.loop.run_until_complete(self.channel.send(text))

File: node/spot.js
const Discord = require('discord.js')
const fs = require('fs')

module.exports = class Spot {
    /**
     * @param {string} id - Discord id of target user for messages.
     * @param {string} token - Bot token.
     * 
     * Usage:
     * spot = new Spot(id, token);
     * spot.message('i love you');
     * spot.exit();
     * 
     */
    constructor({file, id, token}) {
        if (file) {
            const config = fs.readFileSync(file);
            const auth = JSON.parse(config);
            id = auth.id;
            token = auth.token;
        }
        else if (!(id && token)) {
            console.log('spot can not authenticate');
            return
        }

        this.client = new Discord.Client()

        this.user = this.client.login(token).then(() => {
                return this.client.fetchUser(id)
        });
        this.user.then(u => u.send('On Duty!'))
    }

    message(text) {
        this.user.then(u => u.send(text))
    }

    exit() {
        this.user.then(u => u.send('Work finished.'))
            .finally(() => this.client.destroy())
    }
}